### ISSUE TITLE
Cannot delete custom fields - hangs indefinitely after confirm with spinning gear

### ISSUE THREAD

**Issue by wirerydr** [nilsteampassnet/TeamPass#1884]
### Steps to reproduce
1. Install a fresh copy of latest development commit 716c1b8.
2. Create folder **_testfolder_** and role **_testrole_**.
3. Create category **_testcategory_**, and set it to show in folder _**testfolder**_.
4. Inside category _**testcategory**_, create custom field _**testfield**_.
5. Either leave **_Custom item fields enabled_** set to **Yes**, or **No** - it doesn't matter.
5. Select field **_testfield_**, and then click <_**Delete**_>.
6. When the **Confirm** popup appears, click <_**Confirm**_>.

### Expected behaviour
The **Confirm** popup should disappear, and the **Custom Fields** page should be redisplayed with field _**testfield**_ now gone.

### Actual behaviour
The **Confirm** popup remains indefinitely, and the gear continues to spin indefinitely.  The page is not redisplayed.  There is absolutely no output in the Apache logs (error or otherwise).  <_**Confirm**_> can be clicked as many times as you want - nothing changes.
![05_after_confirm_clicked](https://user-images.githubusercontent.com/4572687/29757453-ee400f60-8b79-11e7-8a12-02a10b49b77d.png)
If <_**Cancel**_> is pressed, then the operation cancels properly and the _**Custom Fields**_ page is redisplayed.  the custom-field _**testfield**_ is still present.

### Server configuration
**Operating system**:  Fedora 26 (x86_64)

**Web server:** Apache 2.4.27

**Database:** 10.1.25

**PHP version:** 7.1

**Teampass version:** development commit 716c1b8

**Updated from an older Teampass or fresh install:** Freshly installed

### Client configuration
**Browser:** FireFox 55.0.3 (64-bit)

**Operating system:** Windows 10

#### Web server error log
```
No logging output generated
```

### State of database tables right before clicking <**Confirm**>
![01_teampass_categories](https://user-images.githubusercontent.com/4572687/29757573-d4ca9392-8b7a-11e7-8140-ae4d573c726c.png)
![02_teampass_categories_folders](https://user-images.githubusercontent.com/4572687/29757575-d68b98b6-8b7a-11e7-90c8-2f5edd62b7f3.png)
![03_teampass_categories_items](https://user-images.githubusercontent.com/4572687/29757577-d907ff3a-8b7a-11e7-9ce6-dfec1032d900.png)

**Comment by nilsteampassnet** [comment]
I could reproduce this error.
Thank you for reporting.

Note: Thank you for the good level of details you provide in your tickets. I really appreciate this.

**Comment by nilsteampassnet** [comment]
I will commit the fix just after.

To fix this manually, open file `admin.settings.load.php`, search for
```
                } else if ($("#post_type").val() === "changeFieldType") {
                    data = $("#changeFieldType").val();
                }
                if (data === "") {
                    return false;
                }
```
and replace by
```
                } else if ($("#post_type").val() === "changeFieldType") {
                    data = $("#changeFieldType").val();
                } else if ($("#post_type").val() === "deleteCategory") {
                    data = "no_data";
                }
                if (data === "") {
                    return false;
                }
```

**Comment by wirerydr** [comment]
Tested, and confirmed fixed by development commit 7dd9f6c - thanks for the very quick turnaround.  Closing this issue...

> Thank you for the good level of details you provide in your tickets. I really appreciate this.

Happy to help.  I made sure to drink more coffee this time...
